tests/sysprep_test.go

```diff
@@ -217,161 +215,120 @@ func insertProductKeyToAnswerFileTemplate(answerFileTemplate string) string {
+				Name: "sysprep",
+				VolumeSource: v1.VolumeSource{
+					Sysprep: &v1.SysprepSource{
+						ConfigMap: &k8sv1.LocalObjectReference{
+							Name: cm.Name,
+						},
+					},
+				},
+			}),
+			libvmi.WithResourceMemory("2048Mi"),
+			withFeatures(v1.Features{
+				ACPI: v1.FeatureState{},
+				APIC: &v1.FeatureAPIC{},
+				Hyperv: &v1.FeatureHyperv{
+					Relaxed:   &v1.FeatureState{},
+					VAPIC:     &v1.FeatureState{},
+					Spinlocks: &v1.FeatureSpinlocks{Retries: pointer.P(uint32(8191))},
+				},
+			}),
+			withClock(v1.Clock{
+				ClockOffset: v1.ClockOffset{UTC: &v1.ClockOffsetUTC{}},
+				Timer: &v1.Timer{
+					HPET:   &v1.HPETTimer{Enabled: pointer.P(false)},
+					PIT:    &v1.PITTimer{TickPolicy: v1.PITTickPolicyDelay},
+					RTC:    &v1.RTCTimer{TickPolicy: v1.RTCTickPolicyCatchup},
+					Hyperv: &v1.HypervTimer{},
+				},
+			}),
+			withFirmwareUID(types.UID(libvmifact.WindowsFirmware)),
+		)
 	})
 
 	Context("[ref_id:5105]should create the Admin user as specified in the Autounattend.xml", func() {
 		var winrmcliPod *k8sv1.Pod
-		var cli []string
-		var output string
-		var vmiIp string
 
 		BeforeEach(func() {
 			By("Creating winrm-cli pod for the future use")
-			winrmcliPod = winRMCliPod()
-			winrmcliPod, err = virtClient.CoreV1().Pods(testsuite.NamespaceTestDefault).Create(context.Background(), winrmcliPod, metav1.CreateOptions{})
+			var err error
+			winrmcliPod, err = virtClient.CoreV1().Pods(testsuite.NamespaceTestDefault).Create(context.Background(), winRMCliPod(), metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Starting the windows VirtualMachineInstance")
 			windowsVMI, err = virtClient.VirtualMachineInstance(testsuite.NamespaceTestDefault).Create(context.Background(), windowsVMI, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			libwait.WaitForSuccessfulVMIStart(windowsVMI,
+			windowsVMI = libwait.WaitForSuccessfulVMIStart(windowsVMI,
 				libwait.WithTimeout(720),
 			)
+		})
 
-			windowsVMI, err = virtClient.VirtualMachineInstance(testsuite.NamespaceTestDefault).Get(context.Background(), windowsVMI.Name, metav1.GetOptions{})
-			vmiIp = windowsVMI.Status.Interfaces[0].IP
-			cli = []string{
+		It("[test_id:5843]Should run echo command on machine using the credentials specified in the Autounattend.xml file", func() {
+			command := append([]string{
 				winrmCliCmd,
 				"-hostname",
-				vmiIp,
+				windowsVMI.Status.Interfaces[0].IP,
 				"-username",
 				windowsSysprepedVMIUser,
 				"-password",
 				windowsSysprepedVMIPassword,
-			}
-		})
-
-		It("[test_id:5843]Should run echo command on machine using the credentials specified in the Autounattend.xml file", func() {
-			command := append(cli, "echo works")
-			Eventually(func() error {
-				fmt.Printf("Running \"%s\" command via winrm-cli\n", command)
-				output, err = exec.ExecuteCommandOnPod(
+			}, "echo works")
+			By("Checking that the Windows VirtualMachineInstance has expected UUID")
```

Comment:
How does this check the VM has the expected UUID?

Reply:
Good question, dropping this instead of moving it around.